### PR TITLE
feat(switch): enhances Switch component to allow inline variant in new validation redesign

### PR DESCRIPTION
**Pull request by damienrobson-sage**
### Proposed behaviour

Adds the following enhancements to the `Switch` component:

- Inline error style with the new error 
- Dark background variant

This will allow the inline option to work alongside the `validationRedesignOptIn` flag. If the inline layout does not fit the space provided, it will default to the vertical style layout.

### Current behaviour

The `Switch` component does not currently support inline layouts if the `validationRedesignOptIn` flag is set to true

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [x] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [x] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### CODE REVIEW

File: src/__internal__/checkable-input/checkable-input.style.ts [hunk 1]
```diff
@@ -30,6 +31,8 @@ const StyledCheckableInputWrapper = styled.div<StyledCheckableInputWrapperProps>
     labelInline,
     reverse,
   }) => css`
+    width: 100% !important;
```
[the remark this answers is not in the document]
Comment (damienrobson-sage) [reply src/__internal__/checkable-input/checkable-input.style.ts:34]: Don't look at line 119 then, either 😄

[the remark this answers is not in the document]
Comment (edleeks87) [reply src/__internal__/checkable-input/checkable-input.style.ts:34]: Based on the fact that we already use it lets leave it for now and review later

**Review by edleeks87 — APPROVED** [review]

**Review by mihai-albu-sage — DISMISSED** [review]

**Review by edleeks87 — APPROVED** [review]

File: src/__internal__/validation-message/validation-message.style.ts [hunk 2]
```diff
@@ -2,17 +2,22 @@ import styled, { css } from "styled-components";
...
-    color: ${isWarning
-      ? "var(--colorsSemanticCaution600)"
-      : "var(--colorsSemanticNegative500)"};
-    font-weight: ${isWarning ? "normal" : "500"};
-    margin-top: 0px;
-    margin-bottom: 8px;
-  `}
+  ${({ isWarning, isDarkBackground }) => {
+    const darkBgColour =
+      !isWarning && isDarkBackground
```
Comment (edleeks87) [inline src/__internal__/validation-message/validation-message.style.ts:11]: comment: we can probably remove the `isWarning` check here as it's checked on line 15

**Comment by harpalsingh** [comment]
Looks good to me, wanted to add there is a minor alignment review we need to look at the inline layout but its rendering slightly different in different browsers so that will not impact this ticket.

**Review by mihai-albu-sage — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

File: src/components/switch/switch.style.ts [hunk 3]
```diff
@@ -20,31 +23,55 @@ interface StyledSwitchProps
...
+    reverse,
+    warning,
+    isDarkBackground,
+  }: {
+    reverse: boolean;
+    warning: boolean;
+    isDarkBackground: boolean;
+  }) => {
+    const darkBgColour =
+      !warning && isDarkBackground
```
Comment (edleeks87) [inline src/components/switch/switch.style.ts:45]: question: do we need to check `warning` here as it's checked on line 53?

**Review by edleeks87 — APPROVED** [review]

**Review by mihai-albu-sage — APPROVED** [review]

**Review by dariusbercea-sage — APPROVED** [review]

**Pull request merged**